### ISSUE TITLE
spack install conflates intel-mkl and intel

### ISSUE THREAD

**Issue by sknigh** [LLNL/spack#3925]
``` bash
$ spack install intel-mkl
==> Installing intel-mkl
==> Using cached archive: spack/var/spack/cache/intel-mkl/intel-mkl-2017.2.174.tgz
==> Already staged intel-mkl-2017.2.174-twy6vcyriwtass3cjgsc7wbtkouqnluo in spack/var/spack/stage/intel-mkl-2017.2.174-twy6vcyriwtass3cjgsc7wbtkouqnluo
==> Already patched intel-mkl
==> Building intel-mkl [Package]
==> Found already existing license spack/etc/spack/licenses/intel/license.lic
==> Executing phase : 'install'
==> Error: ProcessError: Command exited with status 2:
    './install.sh' '--silent' 'silent.cfg'
spack/var/spack/repos/builtin/packages/intel/package.py:96, in install:
     74       def install(self, spec, prefix):
     75
     76           if not hasattr(self, "intel_prefix"):
     77               self.intel_prefix = self.prefix
     78
     79           silent_config_filename = 'silent.cfg'
     80           with open(silent_config_filename, 'w') as f:
     81               f.write("""
     82   ACCEPT_EULA=accept
     83   PSET_MODE=install
     84   CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
     85   PSET_INSTALL_DIR=%s
     86   NONRPM_DB_DIR=%s
     87   ACTIVATION_LICENSE_FILE=%s
     88   ACTIVATION_TYPE=license_file
     89   PHONEHOME_SEND_USAGE_DATA=no
     90   CONTINUE_WITH_OPTIONAL_ERROR=yes
     91   COMPONENTS=%s
     92   """ % (self.intel_prefix, self.intel_prefix, self.global_license_file,
     93                   self.intel_components))
     94
     95           install_script = Executable("./install.sh")
  >> 96           install_script('--silent', silent_config_filename)
```

Note that the package it's citing is ```intel``` not ```intel-mkl```

The error output indicates the build script quit upon parsing a cfg intended for the intel compilers.
``` bash
==> './install.sh' '--silent' 'silent.cfg'
Name "ACTIVATION_LICENSE_FILE" is not valid
```

I haven't found other commands that stumble on this.

**Comment by sknigh** [comment]
```spack install mkl``` has the same behavior

**Comment by adamjstewart** [comment]
That's because `IntelMkl` is a subclass of `IntelInstaller` which resides in the `intel` package. All of the intel packages have similar logic so we tried to group it in a single package. Although there's still a lot of duplication.

**Comment by sknigh** [comment]
Is there another way to install MKL, will there need to be changes to ```IntelInstaller``` first?

**Comment by adamjstewart** [comment]
Not sure, I've only ever used the `intel-parallel-studio` package. @lee218llnl @davydden?

**Comment by davydden** [comment]
Intel-mkl works fine for me

**Comment by lee218llnl** [comment]
I believe the IntelInstaller is the only way we currently install MKL.

Based on your error message, it looks like you will need to point the installer to your Intel software license. Did you point Spack to an Intel license file that supports MKL? You can refer to http://spack.readthedocs.io/en/latest/packaging_guide.html#licensed-software for more info. If you haven't done this before, Spack should have prompted you to edit the Spack global license file, but you could also create/edit that file manually.

**Comment by sknigh** [comment]
I doubt the license is an issue, it is the same one I used to install working Intel compilers. I tried installing Intel, then MKL in a clean Centos 7 VM and got the same result.

``` bash
$ spack install intel
==> Installing intel
==> Fetching file:///intel/intel-17.0.2.tgz
######################################################################## 100.0%
==> Staging archive: var/spack/stage/intel-17.0.2-hfm3l7uljieiamfbgyu5iqtgol2gntpc/parallel_studio_xe_2017_update2_composer_edition.tgz
==> Created stage in var/spack/stage/intel-17.0.2-hfm3l7uljieiamfbgyu5iqtgol2gntpc
==> Ran patch() for intel
==> Building intel [Package]
==> Found already existing license etc/spack/licenses/intel/license.lic
==> Executing phase : 'install'
==> Added local symlink opt/spack/linux-centos7-x86_64/gcc-4.8.5/intel-17.0.2-hfm3l7uljieiamfbgyu5iqtgol2gntpc/Licenses/license.lic to global license file
==> Successfully installed intel
  Fetch: 41.28s.  Build: 26m 23.85s.  Total: 27m 5.13s.
[+] opt/spack/linux-centos7-x86_64/gcc-4.8.5/intel-17.0.2-hfm3l7uljieiamfbgyu5iqtgol2gntpc
$ spack install intel-mkl
==> Installing intel-mkl
==> Using cached archive: var/spack/cache/intel-mkl/intel-mkl-2017.2.174.tgz
==> Already staged intel-mkl-2017.2.174-twy6vcyriwtass3cjgsc7wbtkouqnluo in var/spack/stage/intel-mkl-2017.2.174-twy6vcyriwtass3cjgsc7wbtkouqnluo
==> Already patched intel-mkl
==> Building intel-mkl [Package]
==> Found already existing license etc/spack/licenses/intel/license.lic
==> Executing phase : 'install'
==> Error: ProcessError: Command exited with status 2:
    './install.sh' '--silent' 'silent.cfg'
var/spack/repos/builtin/packages/intel/package.py:96, in install:
     74       def install(self, spec, prefix):
     75
     76           if not hasattr(self, "intel_prefix"):
     77               self.intel_prefix = self.prefix
     78
     79           silent_config_filename = 'silent.cfg'
     80           with open(silent_config_filename, 'w') as f:
     81               f.write("""
     82   ACCEPT_EULA=accept
     83   PSET_MODE=install
     84   CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
     85   PSET_INSTALL_DIR=%s
     86   NONRPM_DB_DIR=%s
     87   ACTIVATION_LICENSE_FILE=%s
     88   ACTIVATION_TYPE=license_file
     89   PHONEHOME_SEND_USAGE_DATA=no
     90   CONTINUE_WITH_OPTIONAL_ERROR=yes
     91   COMPONENTS=%s
     92   """ % (self.intel_prefix, self.intel_prefix, self.global_license_file,
     93                   self.intel_components))
     94
     95           install_script = Executable("./install.sh")
  >> 96           install_script('--silent', silent_config_filename)

See build log for details:
  /spack-stage-lx_9rT/l_mkl_2017.2.174/spack-build.out
```

And the same error message
``` bash 
==> './install.sh' '--silent' 'silent.cfg'
Name "ACTIVATION_LICENSE_FILE" is not valid
```

The error message is probably not about the license file itself, but the presence of the tag. A freshly staged ```silent.cfg``` doesn't contain ```ACTIVATION_LICENSE_FILE```

```
# Patterns used to check silent configuration file
#
# anythingpat - any string
# filepat     - the file location pattern (/file/location/to/license.lic)
# lspat       - the license server address pattern (0123@hostname)
# snpat       - the serial number pattern (ABCD-01234567)

# Accept EULA, valid values are: {accept, decline}
ACCEPT_EULA=decline

# Optional error behavior, valid values are: {yes, no}
CONTINUE_WITH_OPTIONAL_ERROR=yes

# Install location, valid values are: {/opt/intel, filepat}
PSET_INSTALL_DIR=/opt/intel

# Continue with overwrite of existing installation directory, valid values are: {yes, no}
CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes

# List of components to install, valid values are: {ALL, DEFAULTS, anythingpat}
COMPONENTS=DEFAULTS

# Installation mode, valid values are: {install, repair, uninstall}
PSET_MODE=install

# Directory for non-RPM database, valid values are: {filepat}
#NONRPM_DB_DIR=filepat

# Path to the cluster description file, valid values are: {filepat}
#CLUSTER_INSTALL_MACHINES_FILE=filepat

# Perform validation of digital signatures of RPM files, valid values are: {yes, no}
SIGNING_ENABLED=yes

# Select target architecture of your applications, valid values are: {IA32, INTEL64, ALL}
ARCH_SELECTED=ALL
```

**Comment by sknigh** [comment]
I noticed ```intel-mkl@11.3.3.210``` has ```ACTIVATION_LICENSE_FILE``` in its freshly staged ```silent.cfg```. I was able to build it too. 

@davydden @lee218llnl have you tried building ```intel-mkl@2017.2.174```?

**Comment by lee218llnl** [comment]
@sknigh, I was able to reproduce the error you reported. It looks like the 2017.2.174 version does not require a license to install and also doesn't like if you specify the ACTIVATION_LICENSE_FILE key. It also doesn't like the ACTIVATION_TYPE and PHONEHOM_SEND_USAGE_DATA keys in silent.cfg. I will work on a fix, which will probably involve having MKL do it's own custom install method for affected versions.

**Comment by lee218llnl** [comment]
The 2017.0.098 version installs with the license file specification. Looks like as of this Feb Intel now offers MKL (and some other libraries) for free https://software.intel.com/en-us/articles/free-mkl.

**Comment by lee218llnl** [comment]
@sknigh This should be addressed by #3931. Can you do a pull on the develop branch and try again?

**Comment by sknigh** [comment]
#3931 resolved my issue. Thank you.
